c/httpserver.c

```diff
@@ -5887,6 +5963,20 @@ int mainHttpLoop(HttpServer *server){
                                             httpWorkElementHandler,
                                             httpBackgroundHandler);
 
+  runMonitoringThread();
+  STCModule *heartbeatMonitoringModule = stcRegisterModule(
+        base,
+        STC_MODULE_GENERIC,
+        server,
+        NULL,
+        NULL,
+        monitoringWorkElementHandler,
+        NULL
```

Comment:
thanks @ifakhrutdinov 
we tried new way in 198, as described above, using STC_GENERIC_MODULE
will move it out of httpserver.c 

monitoring loop will keep track of last user activity
we are planning to run this check every 30sec or 1min
we will delete the locks held by user in semaphore table if last activity of user is found to be too old (30 secs ago)
basically, to hold on to locks users have to keep sending 
GET /datasetHeartbeat - every 30 secs

Reply:
@NakulManchanda 

> we tried new way in 198, as described above, using STC_GENERIC_MODULE
> will move it out of httpserver.c
> monitoring loop will keep track of last user activity

Are you planning to make it generic? If that's used solely for data set, we'll have to rewrite that next time someone needs a background callback.

> we are planning to run this check every 30sec or 1min

I believe that'd be better if we could make this configurable so that users could tune if they want to trade performance for responsiveness.

Reply:
>Are you planning to make it generic? If that's used solely for data set, we'll have to rewrite that next time someone needs a background callback.

need your guidance there, moving discussion to PR 198, refactoring, will send a meeting invite